client/controller/shared.js

```diff
@@ -35,6 +37,62 @@ export function makeLayoutMiddleware( LayoutComponent ) {
 	};
 }
 
+/***
+ * Replaces stylesheet element gracefully
+ *
+ * If we set the `href` attribute of `link` directly, the browser will remove
+ * the previous stylesheet and will start loading the new one, but in the meantime
+ * we won't have all of the sizes the old one had.
+ *
+ * To prevent that, we first load a new stylesheet and then remove the old one
+ *
+ * @param {DOMElement} linkElement Link to change
+ * @param {String} newHref new url to load
+ */
+function replaceStlyesheet( linkElement, newHref ) {
+	const newLink = Object.assign( document.createElement( 'link' ), {
+		rel: 'stylesheet',
+		type: 'text/css',
+		href: newHref,
+	} );
+
+	if ( linkElement.id ) {
+		newLink.id = linkElement.id;
+	}
+
+	function deleteOldLink() {
+		linkElement.parentElement.removeChild( linkElement );
+		newLink.removeEventListener( 'load', deleteOldLink );
+	}
+
+	newLink.addEventListener( 'load', deleteOldLink );
```

Comment:
Did you try it? I don't think link tags emit a `load` event.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#Stylesheet_load_events
https://jsfiddle.net/v26fbb99/2/